Bottom: 616eee86fe606ee581b09ed5bf85d9eac61ddebf
Top:    80eace24936613551af3a4b4a25f3607a8d9dc4b
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-07-17 16:46:41 +0800

Refresh of vhost_net-batch-update-used

---

diff --git a/drivers/vhost/net.c b/drivers/vhost/net.c
index 18c013d..ae87dab 100644
--- a/drivers/vhost/net.c
+++ b/drivers/vhost/net.c
@@ -531,6 +531,10 @@ static int get_tx_bufs(struct vhost_net *net,
 		goto err;
 	}
 
+	vq->heads[nvq->done_idx].id = cpu_to_vhost32(vq, ret);
+	vq->heads[nvq->done_idx].len = 0;
+	++nvq->done_idx;
+
 	return 0;
 err:
 	*len = 0;
@@ -611,7 +615,7 @@ static void handle_tx_copy(struct vhost_net *net)
 		if (err != len)
 			pr_debug("Truncated TX packet: "
 				 " len %d != %zd\n", err, len);
-		if (++nvq->done_idx >= VHOST_NET_BATCH)
+		if (nvq->done_idx >= VHOST_NET_BATCH)
 			vhost_net_signal_used(nvq);
 		if (vhost_exceeds_weight(++sent_pkts, total_len)) {
 			vhost_poll_queue(&vq->poll);
